src/connectors/connectorKAF.js

```diff
@@ -0,0 +1,322 @@
+                    sessionTimeout: 15000,
+                    protocol: ['roundrobin'],
+                    fromOffset: 'latest'
+                }
+                //console.log(consumerOptions);
+                var topic = [this.params.topic];
+
+                var consumerGroup = new kafka.ConsumerGroup(Object.assign({ id: 'BGPAlerter' }, consumerOptions), topic);
+                this.consumer = consumerGroup;
+                this._appendListeners(resolve, reject);
+
+
+            } catch(error) {
+                this._error(error);
+                console.log(error);
+                reject(error);
+            }
+        });
+
+    disconnect = () => {
+        if (this.ws) {
+            this._disconnect(`${this.name} disconnected`);
+        }
+    };
+
+    _subscribeToAll = (input) => {
+        //this.ws.send(JSON.stringify({
+        //    type: "ris_subscribe",
+        //    data: this.params.subscription
+        //}));
+    };
+
+    _optimizedPathMatch = (regex) => {
+
+        if (regex) {
+            regex = (regex.slice(0,2) === ".*") ? regex.slice(2) : regex;
+            regex = (regex.slice(-2) === ".*") ? regex.slice(0,-2) : regex;
+            const regexTests = [
+                "^[\\^]*\\d+[\\$]*$",
+                "^[\\^]*[\\d+,]+\\d+[\\$]*$",
+                "^[\\^]*\\[[\\d+,]+\\d+\\][\\$]*$"
+            ];
+
+            for (let r of regexTests) {
+                if (new RegExp(r).test(regex)) {
+                    return regex;
+                }
+            }
+        }
+
+        return null;
+    };
+
+    _subscribeToPrefixes = (input) => {
+        const monitoredPrefixes = input.getMonitoredLessSpecifics();
+        const params = JSON.parse(JSON.stringify(this.params.subscription));
+
+        if (monitoredPrefixes
+            .filter(
+                i => (ipUtils.isEqualPrefix(i.prefix, '0:0:0:0:0:0:0:0/0') || ipUtils.isEqualPrefix(i.prefix,'0.0.0.0/0'))
+            ).length === 2) {
+
+            delete params.prefix;
+
+            if (!this.subscribed["everything"]) {
+                console.log("Monitoring everything");
+                this.subscribed["everything"] = true;
+            }
+
+        } else {
+            for (let p of monitoredPrefixes) {
+
+                if (!this.subscribed[p.prefix]) {
+                    console.log("Monitoring", p.prefix);
+                    this.subscribed[p.prefix] = true;
+                }
+
+                params.prefix = p.prefix;
+            }
+        }
+    };
+
+    _subscribeToASns = (input) => {
+        const monitoredASns = input.getMonitoredASns().map(i => i.asn);
+        for (let asn of monitoredASns){
+            const asnString = asn.getValue();
+            if (!this.subscribed[asnString]) {
+                console.log(`Monitoring AS${asnString}`);
+                this.subscribed[asnString] = true;
+            }
+        }
+    };
+
+    _onInputChange = (input) => {
+        this.connect()

```

Comment:
It looks like every time the input file changes, a new kafka connection is established without closing the previous one. Am I correct?